armada_command/command_run.py

```diff
@@ -120,64 +160,73 @@ def command_run(args):
+            rename, image.image_name, dockyard_string, ship_string))
     else:
         print('Running microservice {} from dockyard: {}{}...'.format(
             image.image_name, dockyard_string, ship_string))
 
-    payload = {'image_path': image.image_path, 'environment': {}, 'ports': {}, 'volumes': {}}
+
+def _payload_update_dockyard(dockyard_alias, payload):
     if dockyard_alias and dockyard_alias != 'local':
         dockyard_info = dockyard.alias.get_alias(dockyard_alias)
         if not dockyard_info:
             raise ArmadaCommandException("Couldn't read configuration for dockyard alias {0}.".format(dockyard_alias))
-        payload['dockyard_user'] = dockyard_info.get('user')
+        payload['dockyard_user'] = dockyard_info.get('user'),
         payload['dockyard_password'] = dockyard_info.get('password')
 
-    if vagrant_dev:
-        if not args.dynamic_ports:
-            payload['ports']['4999'] = '80'
-            payload['ports']['2999'] = '22'
-        if not args.use_latest_image_code:
-            microservice_path = '/opt/{microservice_name}'.format(**locals())
-            payload['volumes'][microservice_path] = microservice_path
-        payload['environment']['ARMADA_VAGRANT_DEV'] = '1'
-
-    hermes_env, hermes_volumes = process_hermes(image.image_name, args.env, args.app_id,
-                                                sum(args.configs or [], []))
+
+def _payload_update_vagrant(dynamic_ports, latest_image_code, microservice_name, payload):
+    if not dynamic_ports:
+        payload['ports']['4999'] = '80'
+    if not latest_image_code:
+        microservice_path = '/opt/{microservice_name}'.format(**locals())
+        payload['volumes'][microservice_path] = microservice_path
+    payload['environment']['ARMADA_VAGRANT_DEV'] = '1'
+
+
+def _payload_update_hermes(image_name, env, app_id, configs, payload):
+    hermes_env, hermes_volumes = process_hermes(image_name, env, app_id,
+                                                sum(configs or [], []))
     payload['environment'].update(hermes_env or {})
     payload['volumes'].update(hermes_volumes or {})
 
-    # --- environment
-    for env_var in sum(args.e or [], []):
+
+def _payload_update_environment(env_vars, payload):
+    for env_var in sum(env_vars or [], []):
         env_key, env_value = (env_var.strip('"').split('=', 1) + [''])[:2]
         payload['environment'][env_key] = env_value
 
-    # --- ports
-    for port_mapping in sum(args.publish or [], []):
+
+def _payload_update_ports(ports, payload):
+    for port_mapping in sum(ports or [], []):
         try:
             port_host, port_container = map(int, (port_mapping.split(':', 1) + [None])[:2])
             payload['ports'][str(port_host)] = str(port_container)
         except (ValueError, TypeError):
             print('Invalid port mapping: {0}'.format(port_mapping), file=sys.stderr)
-            return
+            sys.exit(1)
```

Comment:
We should raise exception instead and have one place to handle errors, which is: https://github.com/armadaplatform/armada/blob/766ea71c6cc65d291c95f5a48807c414842718c6/armada_command/armada.py#L154